components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java

```diff
@@ -153,7 +153,9 @@ private CarbonOAuthTokenRequest handleInvalidRequest(OAuthProblemException e)
                 log.debug("Error: " + e.getError() + ", description: " + e.getDescription());
             }
         } else {
```

Comment:
We can remove the else block and move the logging outside

Reply:
The  logging has to be different as per the existing logic. Hence, else part is needed IMO.